core/src/main/java/io/parsingdata/metal/expression/value/reference/Offset.java

```diff
@@ -40,10 +40,7 @@
 
     @Override
     public Optional<Value> eval(final Value value, final Environment environment, final Encoding encoding) {
-        if (value instanceof ParseValue) {
-            return Optional.of(ConstantFactory.createFromNumeric(((ParseValue) value).slice.offset, value.encoding));
-        }
-        return Optional.empty();
+        return Optional.of(ConstantFactory.createFromNumeric(value.slice.offset, value.encoding));
```

Comment:
I had the same reaction! So we lose the `instanceof`, the cast _and_ the `empty()`! :)